dune-project

```
@@ -26,6 +26,8 @@
   (cinaps (and :with-test (>= v0.12.1))))
  (conflicts
   (ocaml-migrate-parsetree (< 2.0.0))
+  (ocaml-base-compiler (= 5.1.0~alpha1))
+  (ocaml-variants (= 5.1.0~alpha1+options))
```

Naive question, but why don't you conflict with the `ocaml` package version `5.1.0~alpha1`?

```suggestion
  (ocaml (= 5.1.0~alpha1))
```

---

See https://github.com/ocaml/opam-repository/pull/25267#pullrequestreview-1884982506